datadog_checks_base/datadog_checks/base/utils/db/utils.py

```diff
@@ -182,19 +182,22 @@ def default_json_event_encoding(o):
 
 
 def obfuscate_sql_with_metadata(query, options=None):
-    try:
-        statement = datadog_agent.obfuscate_sql(query, options)
-        statement_with_metadata = json.loads(statement)
-    except ValueError:  # Backwards compatible exception between py38 and py27 for json.loads()
-        # Assume we're running against an older agent and return the obfuscated query without metadata.
-        return {'query': statement, 'metadata': {}}
-    except Exception as e:
-        raise e
-    metadata = statement_with_metadata.get('metadata', {})
-    tables = metadata.pop('tables_csv', None)
-    tables = [table.strip() for table in tables.split(',') if table != ''] if tables else None
-    statement_with_metadata['metadata']['tables'] = tables
-    return statement_with_metadata
+    def _load_metadata(statement):
+        try:
+            statement_with_metadata = json.loads(statement)
+            metadata = statement_with_metadata.get('metadata', {})
+            tables = metadata.pop('tables_csv', None)
+            tables = [table.strip() for table in tables.split(',') if table != ''] if tables else None
+            statement_with_metadata['metadata']['tables'] = tables
+            return statement_with_metadata
+        except ValueError:
+            # Assume we're running against an older agent and return the obfuscated query without metadata.
+            return {'query': statement, 'metadata': {}}
+
+    obfuscated_statement = datadog_agent.obfuscate_sql(query, options)
+    if options and json.loads(options).get('return_json_metadata', False):
+        return _load_metadata(obfuscated_statement)
+    return {'query': obfuscated_statement, 'metadata': {}}
```

Comment:
There is no issue returning a new kind of object here? (rathen than `statement_with_metadata` as before)

Reply:
Yeah, unfortunately it isn't obvious upfront because there's a lot under the hood of `obfuscate_sql`, but the TLDR; `statement_with_metadata` is the same shape, so there is no problem.

Reply:
Thanks for having confirmed